M/MSTM/build_tarballs.jl

```diff
@@ -10,14 +10,32 @@ sources = [
 
 # Bash recipe for building across all platforms
 script = raw"""
-cd $WORKSPACE/srcdir/MSTM/code
-mpifort -O2 mpidefs-parallel.f90 mstm-intrinsics.f90 mstm-v4.0.f90 -o "${bindir}/mstm${exeext}"
+if [[ "$target" == x86_64-w64-mingw32 ]]; then
+    cd $WORKSPACE/destdir/include
+    cp $WORKSPACE/destdir/src/mpi.f90 .
+    gfortran -DWIN64 -DINT_PTR_KIND=8 -fno-range-check mpi.f90 || true
+    cd $WORKSPACE/srcdir/MSTM/code
+    echo "void __guard_check_icall_fptr(unsigned long ptr) { }" > cfg_stub.c
+    gcc -c cfg_stub.c
+    gfortran -O2 -fno-range-check mpidefs-parallel.f90 mstm-intrinsics.f90 mstm-v4.0.f90 cfg_stub.o -L${WORKSPACE}/destdir/lib -I${WORKSPACE}/destdir/include -lmsmpifec64 -lmsmpi64 -o "${bindir}/mstm${exeext}"
```

Comment:
Please always use the most specific [environment variables](https://docs.binarybuilder.org/stable/build_tips/#Automatic-environment-variables) available:
```suggestion
    gfortran -O2 -fno-range-check mpidefs-parallel.f90 mstm-intrinsics.f90 mstm-v4.0.f90 cfg_stub.o -L"${libdir}" -I"${includedir}" -lmsmpifec64 -lmsmpi64 -o "${bindir}/mstm${exeext}"
```
`${WORKSPACE}/destdir/lib` is even wrong since libraries for Windows are under `bin/`

Reply:
> > Is ${libdir} strictly equivalent to ${WORKSPACE}/destdir/lib here?
> 
> No, and that's what I [said above](https://github.com/JuliaPackaging/Yggdrasil/pull/4479#discussion_r811346498)

But the lib files (MPI libs) I need are located in `/lib` instead of `/bin` for Windows. Maybe that was the cause?

Reply:
What do you mean by "MPI libs"?  The shared libraries are in `bin/` (where they should be):
```console
% tar tzvf MicrosoftMPI.v10.1.3.x86_64-w64-mingw32.tar.gz
-rwxr-xr-x 0/0          544536 1970-01-01 01:00 bin/IMB_MPI1.exe
-rwxr-xr-x 0/0          558360 1970-01-01 01:00 bin/IMB_NBC.exe
-rwxr-xr-x 0/0          211224 1970-01-01 01:00 bin/MpiPingpong.exe
-rwxr-xr-x 0/0          131277 1970-01-01 01:00 bin/generate_compile_time_mpi_constants.exe
-rwxr-xr-x 0/0          130578 1970-01-01 01:00 bin/libload_time_mpi_constants.dll
-rwxr-xr-x 0/0          549656 1970-01-01 01:00 bin/mpiexec.exe
-rwxr-xr-x 0/0         1631000 1970-01-01 01:00 bin/msmpi.dll
-rwxr-xr-x 0/0          159512 1970-01-01 01:00 bin/msmpilaunchsvc.exe
-rwxr-xr-x 0/0          443160 1970-01-01 01:00 bin/msmpires.dll
-rwxr-xr-x 0/0          426776 1970-01-01 01:00 bin/smpd.exe
-rw-r--r-- 0/0          175354 1970-01-01 01:00 include/mpi.h
-rw-r--r-- 0/0           20078 1970-01-01 01:00 include/mpif.h
-rw-r--r-- 0/0             266 1970-01-01 01:00 include/mpifptr.h
-rw-r--r-- 0/0             143 1970-01-01 01:00 include/mpio.h
-rw-r--r-- 0/0         1058451 1970-01-01 01:00 include/mpitrace.man
-rw-r--r-- 0/0            5278 1970-01-01 01:00 include/mspms.h
-rw-r--r-- 0/0           17980 1970-01-01 01:00 include/pmidbg.h
-rw-r--r-- 0/0          161382 1970-01-01 01:00 lib/libload_time_mpi_constants.dll.a
-rw-r--r-- 0/0          337216 1970-01-01 01:00 lib/msmpi.lib
-rw-r--r-- 0/0          321236 1970-01-01 01:00 lib/msmpi64.lib
-rw-r--r-- 0/0         1075098 1970-01-01 01:00 lib/msmpifec.lib
-rw-r--r-- 0/0         1020116 1970-01-01 01:00 lib/msmpifec64.lib
-rw-r--r-- 0/0         1147914 1970-01-01 01:00 lib/msmpifes.lib
-rw-r--r-- 0/0         1077274 1970-01-01 01:00 lib/msmpifmc.lib
-rw-r--r-- 0/0         1035480 1970-01-01 01:00 lib/msmpifmc64.lib
-rw-r--r-- 0/0         1147914 1970-01-01 01:00 lib/msmpifms.lib
-rw-r--r-- 0/0            4890 1970-01-01 01:00 logs/MicrosoftMPI/MicrosoftMPI.log.gz
-rw-r--r-- 0/0            1076 1970-01-01 01:00 share/licenses/MicrosoftMPI/LICENSE.md
-rw-r--r-- 0/0           13491 1970-01-01 01:00 share/licenses/MicrosoftMPI/MPI_Redistributables_TPN.txt
-rw-r--r-- 0/0            2150 1970-01-01 01:00 share/licenses/MicrosoftMPI/MPI_SDK_TPN.txt
-rw-r--r-- 0/0          224476 1970-01-01 01:00 share/licenses/MicrosoftMPI/MicrosoftMPI_Redistributable_EULA.rtf
-rw-r--r-- 0/0          224238 1970-01-01 01:00 share/licenses/MicrosoftMPI/MicrosoftMPI_SDK_EULA.rtf
-rw-r--r-- 0/0           55162 1970-01-01 01:00 src/mpi.f90
```
I think the files under `lib/` are import libraries?

Reply:
I need the `.lib` files instead of the `.dll` files to compile the program.

Reply:
Why?  Still, please use `${prefix}` instead of `${WORKSPACE}/destdir`.  No one should ever use `${WORKSPACE}/destdir` directly, we may move the prefix at some point in the future (which we actually did already and had to keep a symlink to not break recipes).

Reply:
Because the original Fortran program does not compile when using dynamic libraries...